File: backend/app/schemas/exercise.py
from datetime import datetime
from pydantic import BaseModel


class ExerciseCreate(BaseModel):
    name: str
    muscle_group: str | None = None


class ExerciseUpdate(BaseModel):
    name: str | None = None
    muscle_group: str | None = None


class ExerciseOut(BaseModel):
    id: int
    name: str
    muscle_group: str | None
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/exercise.ts
export interface ExerciseCreate {
  name: string;
  muscle_group: string | null;
}

export interface ExerciseUpdate {
  name?: string;
  muscle_group?: string | null;
}

export interface ExerciseOut {
  id: number;
  name: string;
  muscle_group: string | null;
  created_at: string;
}
